core/envelope-sim/src/main/java/fr/sncf/osrd/envelope_sim/allowances/AbstractAllowanceWithRanges.java

```diff
@@ -163,23 +166,20 @@ private Envelope[] computeAllowanceRegion(Envelope envelopeRegion, EnvelopeSimCo
             imposedTransitionSpeeds[i] = NaN;
         imposedTransitionSpeeds[ranges.size()] = envelopeRegion.getEndSpeed();
 
-        // build an array of (range, percentage) in order to sort the array rangeOrder by ascending percentage
-        var rangePercentages = new RangePercentage[ranges.size()];
+        // build an array of (range, baseTime) in order to sort the array rangeOrder by ascending base time
+        var baseTimes = new RangeBaseTime[ranges.size()];
         for (var i = 0; i < ranges.size(); i++) {
             var range = ranges.get(i);
-            var percentage = range.value.getAllowanceRatio(
-                    envelopeRegion.getTimeBetween(range.beginPos, range.endPos),
-                    range.endPos - range.beginPos
-            );
-            rangePercentages[i] = new RangePercentage(range, percentage);
+            var baseTime = envelopeRegion.getTimeBetween(range.beginPos, range.endPos);
+            baseTimes[i] = new RangeBaseTime(range, baseTime);
         }
 
         // the order in which the ranges should be computed
-        // ranges are computed with increasing percentage values
+        // ranges are computed with increasing baseTime values
         var rangeOrder = new Integer[ranges.size()];
         for (var i = 0; i < ranges.size(); i++)
             rangeOrder[i] = i;
-        Arrays.sort(rangeOrder, Comparator.comparingDouble(rangeIndex -> rangePercentages[rangeIndex].percentage));
+        Arrays.sort(rangeOrder, Comparator.comparingDouble(rangeIndex -> baseTimes[rangeIndex].baseTime));
```

Comment:
Computing allowances this way makes it not optimum in terms of energy consumption. I get that it would help fix some weird cases but the original idea was to keep the total energy consumption as low as possible. Do we agree to get rid of this behavior ? Isn't there another way to solve the problem ?